multilinear_extensions/src/virtual_polys.rs

```diff
@@ -184,155 +196,139 @@ impl<'a, E: ExtensionField> VirtualPolynomials<'a, E> {
     ///
     /// the per-thread instances are registered locally and stored in `thread_based_mles_storage`
     /// using the MLE’s raw pointer as the key to ensure uniqueness and reference consistency.
-    pub fn register_mles(&mut self, mles: Vec<&'a ArcMultilinearExtension<'a, E>>) {
+    pub fn register_mles(&mut self, mles: Vec<EitherRefMLE<'a, E>>) -> Vec<usize> {
```

Comment:
We need to assert that the input `mles` are distinct. Otherwise, the indices returned is different from the one stored in `VirtualPolynomialBuilder`. For example, you can run the simple test in https://github.com/scroll-tech/ceno/tree/feat/mle_ref_test with 

```
cargo test --release --lib test_register_mles
```

Reply:
```rust
let mut virtual_polys = VirtualPolynomials::<E>::new(num_threads, max_num_variables);
let mle = MultilinearExtension::random(max_num_variables, &mut rng);
let mles = vec![EitherRefMLE::Left(&mle), EitherRefMLE::Left(&mle)];

let indices = virtual_polys.register_mles(mles);

assert_eq!(indices.len(), 1, "indices.len() = {}", indices.len());
```